### ISSUE TITLE
DOC: add links to edit on GitHub

### ISSUE THREAD

**Issue by afeld** [pandas-dev/pandas#39859]
#### Location of the documentation

https://pandas.pydata.org/docs/

#### Documentation problem

I'll write it as a user story:

_As a pandas user, when I come across a small problem with the documentation (like a typo), I want to be able to suggest a fix with minimal friction._

Currently, contributors need to search the repository to find the corresponding source file.

#### Suggested fix for documentation

Having an "Edit this page on GitHub" button would help people contribute to documentation quickly. A couple of resources explaining how to do this with Sphinx:

- https://mg.pov.lt/blog/sphinx-edit-on-github.html
- https://gist.github.com/mgedmin/6052926

**Comment by attack68** [comment]
I like this idea, I think it might work well, especially for typos. But it might also have problems...

The API has a more clearly defined structure, it would be easier to approve or decline changes to API docs subject to that. Changes to non-api pages are more subjective, adding in, deleting or editing a sentence here and there might be more difficult to selectively approve based on the number of new participants, and a more dynamic guidebook may be harder to use for regular users. I am assuming the number of editors picks up substantially.

**Comment by MarcoGorelli** [comment]
> Currently, contributors need to search the repository to find the corresponding source file.

TBH I think that having a bit of friction for making a PR is a good thing - I get the impression that PRs done directly from GitHub (usually made from `patch-1` branches) tend to be of lower quality

**Comment by MarcoGorelli** [comment]
> > Currently, contributors need to search the repository to find the corresponding source file.
> 
> TBH I think that having a bit of friction for making a PR is a good thing - I get the impression that PRs done directly from GitHub (usually made from `patch-1` branches) tend to be of lower quality

Sorry @afeld , I think I'd misunderstood what you were suggesting - the button just brings them to the source code. In which case, doesn't the current `source` link already do that?

![image](https://user-images.githubusercontent.com/33491632/108249519-142f5980-714d-11eb-8e85-e584c42420bb.png)

(as an aside, thanks for all the work you're doing on the docs!)

**Comment by afeld** [comment]
Seems like the two folks who have commented already are less in favor. Anyone else think it's a good idea? If not, fine to close.

> having a bit of friction for making a PR is a good thing

I get what you're saying. Setting up a pandas development is probably an hour minimum — would love a number _closer_ to zero, particularly for working on docs — but that's a whole different problem.

> doesn't the current `[source]` button on the right already do that?

Good point! In that case, talking about the non-API pages.

> thanks for all the work you're doing on the docs!

You're welcome! Definitely the largest open source project I've worked on consistently, so getting a lot out of seeing how that works.

**Comment by MarcoGorelli** [comment]
> Setting up a pandas development is probably an hour minimum

True, but the docs do say that you don't need to build pandas locally if you're only interested in building the docs

> Good point! In that case, talking about the non-API pages.

OK yes, if non-API pages don't have a button to make the source easily findable, then adding one seems like a good idea

**Comment by afeld** [comment]
> the docs do say that you don't need to build pandas locally if you're only interested in building the docs

Beside the point here, but:

- Not seeing where it says that
- I'm not sure that's true, since [Sphinx imports pandas](https://github.com/pandas-dev/pandas/blob/00a622401e06bd7afaaa508707a46f3dcc494fe4/doc/source/conf.py#L163)

**Comment by MarcoGorelli** [comment]
You're right, sorry, it just says

> If you’re making documentation changes, you can skip to Contributing to the documentation but if you skip creating the development environment you won’t be able to build the documentation locally before pushing your changes.

So yeah, +1 for adding a button like [source] to non-API pages

**Comment by attack68** [comment]
> Seems like the two folks who have commented already are less in favor. Anyone else think it's a good idea? If not, fine to close.

I said "I like this idea, I think it might work well, especially for typos". sure I cited some potential probs but only potential - helping users to find the source for the non-api is also useful, so I'm more like a +0.5. :P

**Comment by Komal7209** [comment]
@afeld  As its still mentioned that this issue is still opened, so I want to take up this issue and do as per you could suggest more. I want to contribute for this, do let me know where exact I could find the code base folder where I could add this

**Comment by ADing4818** [comment]
I see that this thread has not been commented on for over 3 months and this issue is still up.

I would like to contribute to this @afeld, do you know where in the codebase I could add this feature?

**Comment by ADing4818** [comment]
take

**Comment by arneja-arnav** [comment]
Is this Issue still open? From what I've read of the thread, the issue has been dealt with by @ADing4818

**Comment by ADing4818** [comment]
> Is this Issue still open? From what I've read of the thread, the issue has been dealt with by @ADing4818

Yes, I have written a solution and have been waiting to be reviewed but the contributors have unfortunately not been responsive these past couple of months.

**Comment by MarcoGorelli** [comment]
Sorry for the delay, just left a review - if I've understood the PR correctly, I don't think it quite addresses the issue

**Comment by rhshadrach** [comment]
@lithomas1 - you tagged this as needs discussion a while ago, is that still the case?

**Comment by lithomas1** [comment]
Seems like there weren't any objections, removing the label.

**Comment by DoNguyenHung** [comment]
I would like to work on this issue if no one else is.

**Comment by Sivayogeith** [comment]
fixed with #61956

**Comment by DoNguyenHung** [comment]
Just added my own fix with #61997

**Comment by mroeschke** [comment]
The implementation needed to implement this functionality in https://github.com/pandas-dev/pandas/pull/61956 makes me weary of maintaining this (there doesn't appear to be a Sphinx-only way to achieve this), so IMO not sure if it's worth having this feature

**Comment by afeld** [comment]
Apologies for never responding earlier.

> there doesn't appear to be a Sphinx-only way to achieve this

Yeah, too bad. If anyone figures out a way that requires minimal custom code in this repository, please do open a pull request, but otherwise going to close.
